sdk/monitor/monitor-ingestion/src/logsIngestionClient.ts

```diff
@@ -62,42 +62,35 @@ export class LogsIngestionClient {
     logs: Record<string, unknown>[],
     // eslint-disable-next-line @azure/azure-sdk/ts-naming-options
     options?: UploadLogsOptions
-  ): Promise<UploadLogsResult> {
+  ): Promise<void> {
     // TODO: Do we need to worry about memory issues when loading data for 100GB ?? JS max allocation is 1 or 2GB
 
     // This splits logs into 1MB chunks
     const chunkArray = splitDataToChunks(logs);
-    const noOfChunks = chunkArray.length;
     const concurrency = Math.max(options?.maxConcurrency ?? DEFAULT_MAX_CONCURRENCY, 1);
 
     const uploadResultErrors: Array<UploadLogsError> = [];
-    await concurrentRun(concurrency, chunkArray, async (eachChunk): Promise<void> => {
-      try {
-        await this._dataClient.upload(ruleId, streamName, eachChunk, {
-          contentEncoding: "gzip",
-        });
-      } catch (e: any) {
-        uploadResultErrors.push({
-          cause: e,
-          failedLogs: eachChunk,
-        });
-      }
-    });
-
-    if (uploadResultErrors.length === 0) {
-      return {
-        status: "Success",
-      };
-    } else if (uploadResultErrors.length < noOfChunks && uploadResultErrors.length > 0) {
-      return {
-        errors: uploadResultErrors,
-        status: "PartialFailure",
-      };
-    } else {
-      return {
-        errors: uploadResultErrors,
-        status: "Failure",
-      };
-    }
+    await concurrentRun(
+      concurrency,
+      chunkArray,
+      async (eachChunk): Promise<void> => {
+        try {
+          await this._dataClient.upload(ruleId, streamName, eachChunk, {
+            contentEncoding: "gzip",
+            abortSignal: options?.abortSignal,
+          });
+        } catch (e: any) {
+          if (options?.onError) {
+            await options.onError({ failedLogs: eachChunk, cause: isError(e) ? e : new Error(e) });
```

Comment:
Per previous comment, I feel like onError should have a `void` return and not be awaited
```suggestion
            options.onError({ failedLogs: eachChunk, cause: isError(e) ? e : new Error(e) });
```